Sources/CryptorECC/ECPrivateKey.swift

```diff
@@ -288,26 +292,33 @@ public class ECPrivateKey {
         } else {
             kAsymmetricCryptoManagerKeySize = 521
         }
-        // parameters
-        let parameters: [String: AnyObject] = [
-            kSecAttrKeyType as String:          kAsymmetricCryptoManagerKeyType,
-            kSecAttrKeySizeInBits as String:    kAsymmetricCryptoManagerKeySize as AnyObject,
-            ]
-        var pubKey, privKey: SecKey?
-        let status = SecKeyGeneratePair(parameters as CFDictionary, &pubKey, &privKey)
-        guard status == 0, let newPubKey = pubKey, let newPrivKey = privKey else {
+        
+        let parameters: [NSObject: Any] = [
+            kSecAttrKeyType: kAsymmetricCryptoManagerKeyType,
+            kSecAttrKeySizeInBits: kAsymmetricCryptoManagerKeySize,
+        ]
+        
+        var error: Unmanaged<CFError>?
+
+        // The keys are automatically stored in the keychain
+        guard let privKey = SecKeyCreateRandomKey(parameters as CFDictionary, &error),
+            let pubKey = SecKeyCopyPublicKey(privKey) else {
             throw ECError.failedNativeKeyCreation
```

Comment:
Should this follow the pattern below of ` error.takeRetainedValue() ... throw error`?

Reply:
That's fair...makes sense to be consistent in the error handling, will make the adjustment and update.